File: app/src/main/java/com/example/waveplayer/ViewModelUserPickedSongs.java
package com.example.waveplayer;


import androidx.annotation.GuardedBy;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.ArrayList;
import java.util.List;

public class ViewModelUserPickedSongs extends ViewModel {

    @GuardedBy("this")
    private final List<Song> userPickedSongs = new ArrayList<>();

    synchronized public List<Song> getUserPickedSongs() {
        return userPickedSongs;
    }

    synchronized public void addUserPickedSong(Song songs) {
        userPickedSongs.add(songs);
    }

    synchronized public void removeUserPickedSong(Song song) {
        userPickedSongs.remove(song);
    }

    synchronized public void clearUserPickedSongs() {
        userPickedSongs.clear();
    }

}

File: app/src/main/java/com/example/waveplayer/ViewModelUserPickedSongs.kt
package com.example.waveplayer

import androidx.annotation.GuardedBy
import androidx.lifecycle.ViewModel
import com.example.waveplayer.random_playlist.Song
import java.util.*

class ViewModelUserPickedSongs constructor() : ViewModel() {
    @GuardedBy("this")
    private val userPickedSongs: MutableList<Song?>? = ArrayList()
    @Synchronized
    fun getUserPickedSongs(): MutableList<Song?>? {
        return userPickedSongs
    }

    @Synchronized
    fun addUserPickedSong(songs: Song?) {
        userPickedSongs.add(songs)
    }

    @Synchronized
    fun removeUserPickedSong(song: Song?) {
        userPickedSongs.remove(song)
    }

    @Synchronized
    fun clearUserPickedSongs() {
        userPickedSongs.clear()
    }
}